src/Libraries/Microsoft.Extensions.Http.Resilience/Internal/RequestMessageSnapshot.cs

```diff
@@ -22,21 +24,40 @@ internal sealed class RequestMessageSnapshot : IResettable, IDisposable
     private Version? _version;
     private HttpContent? _content;
 
-    public static RequestMessageSnapshot Create(HttpRequestMessage request)
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Resilience", "EA0014:The async method doesn't support cancellation", Justification = "Past the point of no cancellation.")]
+    public static async Task<RequestMessageSnapshot> CreateAsync(HttpRequestMessage request)
     {
+        _ = Throw.IfNull(request);
+
         var snapshot = _snapshots.Get();
-        snapshot.Initialize(request);
+        await snapshot.InitializeAsync(request).ConfigureAwait(false);
         return snapshot;
     }
 
-    public HttpRequestMessage CreateRequestMessage()
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Resilience", "EA0014:The async method doesn't support cancellation", Justification = "Past the point of no cancellation.")]
+    public async Task<HttpRequestMessage> CreateRequestMessageAsync()
     {
-        var clone = new HttpRequestMessage(_method!, _requestUri)
+        if (!IsInitialized())
+        {
+            throw new InvalidOperationException($"{nameof(CreateRequestMessageAsync)}() cannot be called on a snapshot object that has been reset and/or has not been initialized");
+        }
+
+        var clone = new HttpRequestMessage(_method!, _requestUri?.OriginalString)
         {
-            Content = _content,
             Version = _version!
         };
 
+        if (_content is StreamContent)
```

Comment:
Not a fan of extra branching, can we keep it to `CloneContentAsync`?

Reply:
If we don't branch, then there could be a performance degradation for existing, non-Streamed content users of the `AddStandardHedgingApi` (i.e. the only current users of this `RequestMessageSnapshot` class).  In fact, I believe you told me to create this extra branch in [your other comment](https://github.com/dotnet/extensions/pull/5112#discussion_r1573985985) on this PR:

> This introduces performance regression for non-streamed content. Create an extra branch for `StreamContent` where this logic belongs.

@martintmk apologies if I am misunderstanding what you mean here in relation to your other comment. Can you please clarify?

Reply:
I mean instead of creating all three branches, we can keep the logic in one. It would keep the code simpler.

Reply:
@martintmk apologies, I'm still not following what you mean by "keep the logic in one [branch]". Can you expand further? Perhaps provide a code example of what you are suggesting?

Reply:
@martintmk ahh nevermind. I believe what you are suggesting here is the same as you are suggesting in [your other comment](https://github.com/dotnet/extensions/pull/5112#discussion_r1576731694):
> ... Can you still try to avoid the extra branches by condensing the code into a single `PrepareContentAsync` method
> It would also reduce the number of mutants you have to kill.

I'm resolving this comment to dedupe the threads. If I'm incorrect and this comment is proposing an entirely separate change, then please feel free to reactivate 👍